wireapp/wire-ios-mono
# fix: create mls group - WPB-6082

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6082" title="WPB-6082" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-6082</a>  [iOS] can't create MLS group with no participants
  </td></table>
  <br />
 




----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Creating a MLS group without users failed

### Causes (Optional)

SelfUser was not added when creating the group

### Solutions

Rework createGroup method to include self user by default.

### Testing

#### Test Coverage (Optional)


#### How to Test

1. enable mls flags
2. choose mls protocol to create a group
3. skip adding participant
it should create the mls group

### Notes (Optional)

- avoid MLS setup to generate an error in case we don't have a client yet
- rework quickSync finish async calls that throw and stop other non related operations

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
